report/sets.go

```diff
@@ -84,6 +86,33 @@ func (s Sets) Lookup(key string) (StringSet, bool) {
 	return MakeStringSet(), false
 }
 
+// LookupMask returns the sets stored under key with using mask and index identifying match.
+func (s Sets) LookupMask(key string, mask string, matchSelect int) (StringSet, bool) {
```

Comment:
I haven't looked very deeply, but I suspect this can be achieved using `ParseAddressNodeID()` rather than a regex; it seems overkill to embed knowledge of the string format _and_ use a regex.
This function is only called in one place (apart from tests), so it doesn't need to be generic.

Reply:
`ParseAddressNodeID()` seems a good approach to me too.